developers_chamber/ecs_utils.py

```diff
@@ -193,25 +193,64 @@ def start_service(cluster, service, count, region):
     except ecs_client.exceptions.ClusterNotFoundException:
         raise ClickException("Cluster not found: '{}'".format(cluster))
     except ecs_client.exceptions.ServiceNotFoundException:
-        raise ClickException("Service not found: '{}'".format(cluster))
+        raise ClickException("Service not found: '{}'".format(service))
     except ClientError as ex:
         raise ClickException(ex)
 
 
-def start_services(cluster, count, region):
+def start_services(cluster, services, count, region):
+    ecs_client = _get_ecs_client(region)
+
+    if count is not None and count < 1:
+        raise ClickException("Count must be greater than zero or 'None' to be determined from autoscaling targets.")
+
+    for service in services:
+        count_per_service = count
+        # service = service.split('/')[1]
+        if count_per_service is None:
+            try:
+                count_per_service = get_min_capacity_for_service(cluster=cluster, service=service, region=region)
```

Comment:
Tohle by se asi dalo nejspíš taky zapsat nějak následovně:
`count_per_service = get_min_capacity_for_service(cluster=cluster, service=service, region=region) if not count else count`
nebo
`count_per_service = count if count else get_min_capacity_for_service(cluster=cluster, service=service, region=region)`,
pak by to mělo dva přínosy:
- před try blokem by nemusela být ta podmínka
- nepřiřazoval bys v každém cyklu count do count_per_service
Tedy za předpokladu, že dále v kódu není ten count/count_per_service někde na pozadí změněn.

Reply:
chápu, že by se dal do try bloku dát přímo if, ale potom mi to přijde méně přehledné

v try bloku mám radši jen kód co opravdu může vyhodit exception, přijde mi to tak srozumitelnější.

Navíc opět count může validně nabývat hodnoty None, tzn podle tebe by to vypadalo nějak takhle:

```
for service in services:
        try:
            count_per_service = get_min_capacity_for_service(cluster=cluster, service=service, region=region) if count is not None else count
        except ClickException as ex:
            raise ClickException(
                'Set explicit count or set up autoscaling with minimum capacity.\n{}'.format(ex)
            )

```

vs moje zpřehledněné řešení

```
for service in services:
        if count_per_service is None:
            try:
                count_per_service = get_min_capacity_for_service(cluster=cluster, service=service, region=region)
            except ClickException as ex:
                raise ClickException(
                    'Set explicit count or set up autoscaling with minimum capacity.\n{}'.format(ex)
                )
        else:
            count_per_service = count
```


nechme hlasovat i @matllubos @radimsuckr

Reply:
Já jsem za tvou variantu @trebidav. Taky se mi víc líbí mít v `try` bloku jen kód, od kterého tu výjimku skutečně čekám.

Reply:
Chápu, no ale pokud bys to chtěl podle tebe, pak by začátek té podmínky musel být `if count is None`, jinak pořád musíš napřed do `count_per_service` přiřadit tu hodnotu `count`, což mi přijde zbytečné...

Reply:
Ad "Taky se mi víc líbí mít v try bloku jen kód, od kterého tu výjimku skutečně čekám." - to chápu  (sice je to delší...), ale přiřazení do proměnné nejspíš výjimku nevyhodí a ta kontrola podmínky na `None` asi taky ne. Primárně mi ale šlo o to vyřazení přiřazení do proměnné na začátku, které je v "polovině" případů nadbytečným krokem, který nic nedělá (když se to pak stejně při spklnění té podmínky v některých případech přepíše).

Reply:
Jinak pozor na tu podmínku v mojí variantě, žřekl bych, žes jí přepsal obráceně (is neco versus is not neco).